src/Lean/Syntax.lean

```diff
@@ -172,6 +172,24 @@ partial def getTailWithPos : Syntax → Option Syntax
   | node _ args         => args.findSomeRev? getTailWithPos
   | _                   => none
 
+open SourceInfo in
+/-- Splits an `ident` into its dot-separated components while preserving source info.
+For example, `` `foo.bla.boo `` ↦ `` [`foo, `bla, `boo] ``. -/
+def identComponents : Syntax → List Syntax
+  | ident info rawStr val _ =>
+    rawStr.splitOn "." |>.map fun ss =>
```

Comment:
Same issue in `resolveName'`, but this can break in presence of escaped identifiers, right? Another reason not to implement it twice :) .

Reply:
If the escaped part includes dots, yes. Is there ever a legitimate use case for this? I would suggest to just forbid dots between French quotes.

Reply:
@Vtec234 Isn't the point of French quotes to specifically get around any limitations the syntax might impose? So would that kind of defeat the point? Also, dots can easily appear when dealing with autogenerated syntax kind names. For example: `notation "...." => id` will have the syntax kind "term....".

Reply:
Not any -- you cannot quote `»` :) But the dots notation is a good counterexample, and we shouldn't forbid it. Thanks.

Reply:
Yeah, I know. :) I meant to say "particularly for ASCII", but I forgot to. XD There is a part of me that wonders though why there is not a way to escape "»" -- as then it would match the semantics of the `Name` object itself (any string). Alternatively, since it can't be quoted I am surprised "»" is not used instead of the underscore for what are intended to be inaccessible names (ex. `_private` or `_hyg` could be `»private` or `»hygiene`).